src/SurfaceFluxes.jl

```diff
@@ -54,186 +65,403 @@ Surface flux conditions, returned from `surface_conditions`.
 
 $(DSE.FIELDS)
 """
-struct SurfaceFluxConditions{FT, VFT}
+struct SurfaceFluxConditions{FT}
     L_MO::FT
-    wθ_flux_star::FT
-    flux::VFT
-    x_star::VFT
-    C_exchange::VFT
+    shf::FT
+    lhf::FT
+    ustar::FT
+    Cd::FT
+    Ch::FT
 end
 
 function Base.show(io::IO, sfc::SurfaceFluxConditions)
     println(io, "----------------------- SurfaceFluxConditions")
-    println(io, "L_MO           = ", sfc.L_MO)
-    println(io, "wθ_flux_star = ", sfc.wθ_flux_star)
-    println(io, "flux           = ", sfc.flux)
-    println(io, "x_star         = ", sfc.x_star)
-    println(io, "C_exchange     = ", sfc.C_exchange)
+    println(io, "L_MO                   = ", sfc.L_MO)
+    println(io, "Sensible Heat Flux     = ", sfc.shf)
+    println(io, "Latent Heat Flux       = ", sfc.lhf)
+    println(io, "u_star                 = ", sfc.ustar)
+    println(io, "C_drag                 = ", sfc.Cd)
+    println(io, "C_heat                 = ", sfc.Ch)
     println(io, "-----------------------")
 end
 
-function surface_fluxes_f!(F, x, nt)
-    param_set = nt.param_set
-    wθ_flux_star = nt.wθ_flux_star
-    z_in = nt.z_in
-    z_0 = length(nt.z_0) > 1 ? Tuple(nt.z_0) : nt.z_0
-    x_in = Tuple(nt.x_in)
-    x_s = Tuple(nt.x_s)
-    n_vars = nt.n_vars
-    scheme = nt.scheme
-    universal_func = nt.universal_func
-    θ_scale = nt.θ_scale
+abstract type AbstractSurfaceLocation end
+struct FirstInterior <: AbstractSurfaceLocation end
+struct Surface <: AbstractSurfaceLocation end
+
+abstract type AbstractSurfaceConditions end
+
+struct Fluxes{VI, VS, SHF, LHF, AUF, RM, RB, MOI} <: AbstractSurfaceConditions
+    vals_in::VI
+    vals_sfc::VS
```

Comment:
```suggestion
struct Fluxes{FT, SHF, LHF, AUF, RM, RB, MOI} <: AbstractSurfaceConditions
    vals_in::ValuesFirstInterior{FT}
    vals_sfc::ValuesSurface{FT}
```
We can do this (on the GPU) so long `ValuesSurface` and `ValuesFirstInterior` are concrete types.

IIUC, we can further simplify this to:

```julia
struct Fluxes{FT, AUF} <: AbstractSurfaceConditions
    vals_in::ValuesFirstInterior{FT}
    vals_sfc::ValuesSurface{FT}
    shf::FT
    lhf::FT
    uf::AUF
    z0m::FT
    z0b::FT
    L_MO_init::FT
end
```
Assuming those other properties are all floats

Reply:
We can apply this same pattern to other concrete `AbstractSurfaceConditions` structs.